src/indexes.rs

```diff
@@ -875,6 +875,169 @@ impl Index {
+    /// `documents` = A slice of documents
+    /// `batch_size` = Optional parameter that allows you to specify the size of the batch
+    /// `batch_size` is 1000 by default
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use serde::{Serialize, Deserialize};
+    /// use meilisearch_sdk::{client::*, document::*};
+    ///
+    /// #[derive(Serialize, Deserialize, Debug)]
+    /// struct Movie {
+    ///     name: String,
+    ///     description: String,
+    /// }
+    ///
+    /// impl Document for Movie {
+    ///     type UIDType = String;
+    ///
+    ///     fn get_uid(&self) -> &Self::UIDType {
+    ///         &self.name
+    ///     }
+    /// }
+    ///
+    /// # futures::executor::block_on(async move {
+    /// let client = Client::new("http://localhost:7700", "masterKey");
+    /// let movie_index = client.index("add_documents_in_batches");
+    ///
+    /// let tasks = movie_index.add_documents_in_batches(&[
+    ///  Movie {
+    ///         name: String::from("Interstellar"),
+    ///         description: String::from("Interstellar chronicles the adventures of a group of explorers who make use of a newly discovered wormhole to surpass the limitations on human space travel and conquer the vast distances involved in an interstellar voyage.")
+    ///  },
+    ///  Movie {
+    ///         // note that the id field can only take alphanumerics characters (and '-' and '/')
+    ///         name: String::from("MrsDoubtfire"),
+    ///         description: String::from("Loving but irresponsible dad Daniel Hillard, estranged from his exasperated spouse, is crushed by a court order allowing only weekly visits with his kids. When Daniel learns his ex needs a housekeeper, he gets the job -- disguised as an English nanny. Soon he becomes not only his children's best pal but the kind of parent he should have been from the start.")
+    ///  },
+    ///  Movie {
+    ///         name: String::from("Apollo13"),
+    ///         description: String::from("The true story of technical troubles that scuttle the Apollo 13 lunar mission in 1971, risking the lives of astronaut Jim Lovell and his crew, with the failed journey turning into a thrilling saga of heroism. Drifting more than 200,000 miles from Earth, the astronauts work furiously with the ground crew to avert tragedy.")
+    ///     }],
+    ///     Some(1),
+    ///     Some("name")
+    /// ).await.unwrap();
+    ///
+    /// client.wait_for_task(tasks.last().unwrap(), None, None).await.unwrap();
+    ///
+    /// let movies = movie_index.get_documents::<Movie>(None, None, None).await.unwrap();
+    /// assert!(movies.len() >= 3);
+    /// # movie_index.delete().await.unwrap().wait_for_completion(&client, None,
+    /// None).await.unwrap();
+    /// # });
+    /// ```
+    pub async fn add_documents_in_batches<T: Document>(
+        &self,
+        documents: &[T],
+        batch_size: Option<usize>,
+        primary_key: Option<&str>,
+    ) -> Result<Vec<Task>, Error> {
```

Comment:
```suggestion
    pub async fn add_documents_in_batches<T: Document>(
        &self,
        documents: impl IntoIterator<Item = T>,
        batch_size: Option<usize>,
        primary_key: Option<&str>,
    ) -> Result<Vec<Task>, Error> {
```

Hey, since you're only iterating on the documents wouldn't this work and be easier to use?

Reply:
So something like? 
```rust
    pub async fn add_documents_in_batches<T: Document>(
        &self,
        documents: impl IntoIterator<Item = T>,
        batch_size: Option<usize>,
        primary_key: Option<&str>,
    ) -> Result<Vec<Task>, Error> {
            let documents: Vec<T> = documents.into_iter().collect();
           // unchanged
```
Just changing the type wouldn't work as methods `len` and `chunks` don't seem to be available for `Iterator`. Making `documents` trait bound would require us to collect it into a `Vec<T>`.

Reply:
Ah, sorry, I thought we had the `chunks` methods available directly on iterators. My bad!
Let's keep your code as-is then I don't want to allocate a full vector with all the documents in the back of the users :+1:

Reply:
Okay. Will rebase this once #261 is merged!

Reply:
Thanks you're the best :grin: